tools/cockpit.spec

```diff
@@ -82,18 +82,54 @@ BuildRequires: krb5-server
 # For documentation
 BuildRequires: xmlto
 
+# == Mandatory components of cockpit == #
+
+# Nothing will work without these sub-packages
 Requires: %{name}-bridge = %{version}-%{release}
-Requires: %{name}-networkmanager = %{version}-%{release}
-Requires: %{name}-ws = %{version}-%{release}
 Requires: %{name}-shell = %{version}-%{release}
-Requires: %{name}-storaged = %{version}-%{release}
-%ifarch x86_64 armv7hl
-Requires: %{name}-docker = %{version}-%{release}
-%endif
+
 %if 0%{?rhel}
 Requires: %{name}-subscriptions = %{version}-%{release}
 %endif
 
+# == Optional components == #
+
+# Fedora 24+ supports "soft" dependencies
+%if 0%{?fedora} >= 24 || 0%{?rhel} >= 8
+
+# Recommends: are components that *should* be installed
+# by default, but may be omitted if desired. Removing
+# a Recommended sub-package from the system will not
+# remove the cockpit meta-package.
+Recommends: %{name}-ws = %{version}-%{release}
+Recommends: %{name}-networkmanager = %{version}-%{release}
+Recommends: %{name}-storaged = %{version}-%{release}
+
+%ifarch x86_64 armv7hl
+Recommends: %{name}-docker = %{version}-%{release}
+%endif #ifarch x86_64 armv7hl
+
+# Suggests: are components that may provide additional
+# functionality in some environments, but are not common
+# enough to be installed by default. Packaging tools may
+# present these as a list of optional things the user
+# may wish to install.
+Suggests: %{name}-pcp = %{version}-%{release}
+Suggests: %{name}-kubernetes = %{version}-%{release}
```

Comment:
I don't think cockpit-kubernetes should be part of the Suggests. It requires configuration to make it work. Perhaps once we've solved https://trello.com/c/2qwR9Kvj/202-login-to-kubernetes-origin it could be.